brping/ping1d.py

```diff
@@ -70,9 +73,11 @@ def __init__(self, device_name, baudrate=115200):
     # @return A new PingMessage: as soon as a message is parsed (there may be data remaining in the buffer to be parsed, thus requiring subsequent calls to read())
     # @return None: if the buffer is empty and no message has been parsed
     def read(self):
-        while self.iodev.in_waiting:
-            b = self.iodev.read()
+        bytes = self.iodev.read(self.iodev.in_waiting)
+        self._input_buffer.extendleft(bytes)
 
+        while len(self._input_buffer):
+            b = chr(self._input_buffer.pop())
```

Comment:
Is `chr` necessary if we cast it to `ord` in the next line ?

Reply:
good catch, that simplifies things.